File: 백준/Gold/1011. Fly me to the Alpha Centauri/Fly me to the Alpha Centauri.py
import sys
input = sys.stdin.readline

t = int(input())
for _ in range (t):
  x, y = map(int, input().split())
  dis = y - x
  n = int(dis ** 0.5)
  if n**2 + n > dis:
    n -= 1

  rest = dis - (n**2 + n)

  if rest == 0:
    print(2*n)

  elif rest <= n+1:
    print(2*n + 1)

  else:
    print(2*n + 2)

File: 백준/Gold/1011. Fly me to the Alpha Centauri/Fly me to the Alpha Centauri.java
import java.io.*;
import java.util.*;

public class Main {
    
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int T = Integer.parseInt(br.readLine());
        for (int tc = 1; tc <= T; tc++) {
        	StringTokenizer st = new StringTokenizer(br.readLine());
        	int x = Integer.parseInt(st.nextToken());
        	int y = Integer.parseInt(st.nextToken());
        	int dis = y - x;
        	int n = (int) Math.pow(dis, 0.5);
        	if (Math.pow(n, 2) + n > dis)
        		n -= 1;
        	
        	double rest = dis - (Math.pow(n, 2) + n);
        	
        	if (rest == 0) System.out.println(2*n);
        	else if (rest <= n + 1) System.out.println(2*n + 1);
        	else System.out.println(2 * n + 2);
        }
    }
}
